StripeConnect/StripeConnectTests/Internal/Webview/ConnectComponentWebViewTests.swift

```diff
@@ -13,30 +13,96 @@ import WebKit
 import SafariServices
 
 class ConnectComponentWebViewTests: XCTestCase {
-    func testSetup() {
+    @MainActor
+    func testFetchClientSecret() async throws {
+        let expectation = XCTestExpectation(description: "Client secret is fetched")
         let componentManager = EmbeddedComponentManager(apiClient: .init(publishableKey: "test"), fetchClientSecret: {
-            XCTFail("Client secret should not be retrieved in this test")
+            expectation.fulfill()
             return ""
         })
-        let webView = ConnectComponentWebView(componentManager: componentManager, componentType: .payouts)
+        let webView = ConnectComponentWebView(componentManager: componentManager, componentType: .payouts, loadContent: false)
         
-        XCTAssertEqual(webView.url, URL(string: "https://connect-js.stripe.com/v1.0/ios_webview.html#component=payouts&publicKey=test"))
+        try await webView.evaluateMessageWithReply(name: "fetchClientSecret",
+                                                   json: "{}")
+        
+        await fulfillment(of: [expectation], timeout: 1.0)
     }
     
     @MainActor
-    func testFetchClientSecret() async throws {
-        let expectation = XCTestExpectation(description: "Client secret is fetched")
+    func testFetchInitParams() async throws {
+        let expectation = self.expectation(description: "Message received")
+        let message = FetchInitParamsMessageHandler.Reply(locale: "fr-FR")
         let componentManager = EmbeddedComponentManager(apiClient: .init(publishableKey: "test"), fetchClientSecret: {
```

Comment:
🚫 Blocker:
Can we also test that passing an appearance on init gets correctly sent to the webview on `fetchInitParams`?